src/inbox.rs

```diff
@@ -566,7 +563,10 @@ impl<A> WaitingSender<A> {
     }
 }
 
-impl<A> Future for WaitingSender<A> {
+impl<M> Future for WaitingSender<M>
+where
+    M: Unpin,
```

Comment:
Is this bound going to limit what messages users can send?

Reply:
Potentially although I expect most messages to be composites of basic data types which are all `Unpin.

Reply:
Hm. I feel like it should be possible to remove this bound, though

Reply:
I will try!

Reply:
You were right, we can remove it!

Reply:
Awesome!